index.js

```diff
@@ -320,13 +325,93 @@ module.exports = {
     return options;
   },
 
+  _shouldHandleTypeScript(config) {
+      let emberCLIBabelConfig = config['ember-cli-babel'] || {};
+      if (typeof emberCLIBabelConfig.enableTypeScriptTransforms === 'boolean') {
+        return emberCLIBabelConfig.enableTypeScriptTransforms;
+      }
+      let typeScriptAddon = this.parent.addons
+        && this.parent.addons.find(a => a.name === 'ember-cli-typescript');
+      return typeof typeScriptAddon !== 'undefined'
+        && semver.gte(typeScriptAddon.pkg.version, '4.0.0-alpha.0');
+  },
+
+	_buildClassFeaturePluginConstraints(constraints, config) {
+    // With older versions of ember-cli-typescript, class feature plugins like
```

Comment:
Can you tweak this comment to indicate specific versions that it applies to? I'm mostly looking to ensure that we can more easily know when the logic can be removed (e.g. after XYZ months from the last release that required it, or X major versions, or something like that but we need the specific versioning to be able to figure that out).